internal/extproc/chatcompletion_processor.go

```diff
@@ -204,11 +222,18 @@ func (c *chatCompletionProcessor) ProcessResponseBody(_ context.Context, body *e
 	c.costs.InputTokens += tokenUsage.InputTokens
 	c.costs.OutputTokens += tokenUsage.OutputTokens
 	c.costs.TotalTokens += tokenUsage.TotalTokens
+	// Track the token usage for metrics
+	c.metrics.TokensTotal.WithLabelValues(c.modelName, "total").Add(float64(tokenUsage.TotalTokens))
+	c.metrics.TokensTotal.WithLabelValues(c.modelName, "prompt").Add(float64(tokenUsage.InputTokens))
+	c.metrics.TokensTotal.WithLabelValues(c.modelName, "completion").Add(float64(tokenUsage.OutputTokens))
+
 	if body.EndOfStream && len(c.config.requestCosts) > 0 {
 		resp.DynamicMetadata, err = c.maybeBuildDynamicMetadata()
 		if err != nil {
 			return nil, fmt.Errorf("failed to build dynamic metadata: %w", err)
 		}
+		c.metrics.RequestsTotal.WithLabelValues(c.backendName, c.modelName, "success").Inc()
```

Comment:
Shouldn't we increment the errors on every error returned by this method and any other processor method as well?
Otherwise, depending on "when" during the lifecycle of the request the failure is, you get error metrics or just nothing.
You can use the same "defer" pattern as suggested above.

Reply:
updated